docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
 - `systemd`, running inside the mini-OS context, will launch the `kata-agent` in
   the same context.
 - The agent will create a new confined context to run the specified command in
   (`date` in this example).
 - The agent will then execute the command (`date` in this example) inside this
-  new context, first setting the root filesystem to the expected Ubuntu\* root
+  new context, first setting the root filesystem to the expected Ubuntu root
   filesystem.
 
 #### Initrd image
 
-A compressed `cpio(1)` archive, created from a rootfs which is loaded into memory and used as part of the Linux startup process. During startup, the kernel unpacks it into a special instance of a `tmpfs` that becomes the initial root filesystem.
+The initrd image is a compressed `cpio(1)` archive, created from a
+rootfs which is loaded into memory and used as part of the Linux
+startup process. During startup, the kernel unpacks it into a special
+instance of a `tmpfs` mount that becomes the initial root filesystem.
+
+The only service running in the context of the initrd is the
+[Agent](#agent) as the init daemon. The real workload the user wishes
+to run is created using libcontainer, creating a container in the same
+manner that is done by `runc`.
+
+| Process | Environment | Notes |
+|-|-|-|
+| [Agent](#agent) | VM root | Runs as the init daemon (PID 1) |
+| container workload | VM container | Managed by the agent |
+
+See also the [process overview](#process-overview).
+
+## Hypervisor
 
-The only service running in the context of the initrd is the [Agent](#agent) as the init daemon. The real workload the user wishes to run is created using libcontainer, creating a container in the same manner that is done by `runc`.
+The [configured](#configuration) [hypervisor](../hypervisors.md) creates
+a VM to host the [agent](#agent) and the [workload](#workload).
+
+> **Note:**
+>
+> The hypervisor process run inside a separate "partial container"
+> environment comprised of cgroups and a separate network namespace
+> from the host.
 
 ## Agent
 
-[`kata-agent`](../../src/agent) is a process running in the guest as a supervisor for managing containers and processes running within those containers.
+The Kata Containers agent ([`kata-agent`](../../src/agent)), written
+in the [Rust programming language](https://www.rust-lang.org), is a
+long running process that runs inside the VM. It acts as the
+supervisor for managing the containers and the [workload](#workload)
+running within those containers.
+
+> **Note:**
+>
+> Only a single agent process is run for each VM created.
+
+### Agent communications protocol
+
+The agent communicates with the other Kata components using a
+[`ttRPC`](https://github.com/containerd/ttrpc-rust) based
+[protocol](../../src/agent/protocols/protos).
+
+See also the [agent control tool](#agent-control-tool).
```

Comment:
This tells users to go see the agent control tool but doesn't state what this is. I think you could combine this with the sentence above and add another sentence explaining what an agent control tool is exactly and then link to i.